app/controllers/static_contents_controller.rb

```diff
@@ -73,6 +72,33 @@ def destroy
     end
   end
 
+  # Helper method to get the right params for the right context, if:
+  # 1) user clicks on type filter -> preserve sorting of id or name
+  # 2) user clicks on id -> sort for id[asc/desc], preserve type, delete name sorting
+  # 3) user clicks on name -> sort for name[asc/desc], preserve type, delete id sorting
+  #
+  # TODO: Write tests for this method (?)
+  def params_for_link(link_type, type=nil)
+    p = Hash.new
+
+    if link_type == :type
+      p[:name] = params[:name] if params[:name].present?
+      p[:id] = params[:id] if params[:id].present?
+      p[:type] = type if type.present?
+    else
+      p[:type] = params[:type] if params[:type].present?
+
+      if link_type == :name
+        p[:name] = params[:name] == 'desc' ? 'asc' : 'desc'
+      elsif link_type == :id
+        p[:id] = params[:id] == 'desc' ? 'asc' : 'desc'
+      end
+    end
+
+    p
```

Comment:
did i mention already somewhere, that i have difficulties with the `p` here. this seems like `puts` for me. could there be a more meaningful variable name?